src/config/builder.rs

```diff
@@ -207,6 +207,7 @@ impl From<Config> for ConfigBuilder {
             secret,
             graceful_shutdown_duration,
             hash: _,
+            ..
```

Comment:
Could this cause the data dir lock to be dropped when reloading the config?

Reply:
Yes, that's the intention here. To unlock and consume the config and then a new lock will be acquired on `build()`.